### PR TITLE
[meshcop] use IEEE EUI-64 rather than Joiner ID on user input

### LINKED ISSUE #2254: EUI64 ? SHA256 of EUI64 : while adding joiner
As per the section 8.10.1.9 of thread specification, User will feed the EUI64 of the device to the Commissioner for adding a thread product. But in native commissioner code of OT, we are mem-comparing the user's EUI64 address against SHA256 of EUI64. 

Code @ Commissioner::HandleRelayReceive API of commissioner.cpp file.

If my understanding is wrong, then user should feed SHA256 of Device's EUI64. What's the right one?

### PR DESCRIPTION
**Pull request by jwhui**
Resolves #2254.

### CODE REVIEW

File: src/cli/README.md [hunk 1]
```diff
@@ -31,6 +31,7 @@ OpenThread test scripts use the CLI to execute test cases.
 * [ipaddr](#ipaddr)
 * [ipmaddr](#ipmaddr)
 * [joiner](#joiner-start-pskd-provisioningurl)
+* [joinerid](#joinerid)
```
Comment (librasungirl) [inline src/cli/README.md]: removed cli hashmacaddr from above L29 ?

File: tools/harness-thci/OpenThread.py [hunk 2]
```diff
@@ -1961,16 +1961,21 @@ def scanJoiner(self, xEUI='*', strPSKd='threadjpaketest'):
...
         timeout = 500
 
-        cmd = 'commissioner joiner add %s %s %s' % (JoinerHashMac, strPSKd, str(timeout))
+        if not isinstance(xEUI, str):
+            eui64 = self.__convertLongToString(xEUI)
+
+            # prepend 0 at the beginning
+            if len(eui64) < 16:
+                eui64 = eui64.zfill(16)
+                print address64
```
Comment (librasungirl) [inline tools/harness-thci/OpenThread.py]: address64->eui64

**Comment by jwhui** [comment]
@Vyrastas, any docs around the use of the commissioning CLIs will need to be updated to reflect this change.  In summary, rather than using the hashmacaddr, we are using the factory-assigned IEEE EUI-64 directly.

**Comment by Vyrastas** [comment]
@jwhui So instead of `hashmacaddr` use `joinerid`?

**Comment by jwhui** [comment]
@Vyrastas, the difference lies in the `commissioner joiner add <addr> <psk>` command.  In particular, `<addr>` is now the output of the `eui64` command, rather than the `hashmacaddr` command.

**Pull request merged**